src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/SelfcareApiServiceImpl.scala

```diff
@@ -80,28 +79,26 @@ final case class SelfcareApiServiceImpl(
 
   override def getUser(userId: String)(implicit
     contexts: Seq[(String, String)],
-    toEntityMarshallerUserInfo: ToEntityMarshaller[UserInfo],
+    toEntityMarshallerUserInfo: ToEntityMarshaller[TenantUser],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem]
   ): Route = {
     logger.info(s"Retrieving user $userId")
     val headers: List[HttpHeader] = headersFromContext()
 
-    val result: Future[UserInfo] = for {
+    val result: Future[TenantUser] = for {
       organizationId <- getOrganizationIdFutureUUID(contexts)
       personId       <- getUidFutureUUID(contexts)
       userUuid       <- userId.toFutureUUID
       selfcareUuid   <- getSelfcareIdFutureUUID(contexts)
-      usersInfoApi   <- selfcareV2ClientService
+      users          <- selfcareV2ClientService
         .getInstitutionProductUsers(
           institutionId = selfcareUuid,
           userId = userUuid.some,
           requesterId = personId,
           roles = Seq.empty
         )
-        .map(_.map(_.toApi(organizationId)))
-
-      users    <- usersInfoApi.traverse(_.toFuture)
-      userInfo <- users.headOption.toFuture(UserNotFound(selfcareUuid, userUuid))
+      usersInfo      <- Future.traverse(users)(_.toApi(organizationId).toFuture)
```

Comment:
same, avoid leveraging on threads here
```suggestion
      usersInfo      <- users.traverse(_.toApi(organizationId)).toFuture
```